### ISSUE TITLE
Azure Active Directory Graph Deprecation -> Microsoft Graph API

### ISSUE THREAD

**Issue by Risc-Skydoo** [BookStackApp/BookStack#3028]
### Describe the Bug

I am unable to get AzureAD login working. 

After some research, I found that the Azure Active Directory Graph API will be deprecated in June 2022, and I am unable to activate it in the Azure admin panel. 

More info here :[Update your applications to use Microsoft Authentication Library and Microsoft Graph API](https://techcommunity.microsoft.com/t5/azure-active-directory-identity/update-your-applications-to-use-microsoft-8-library/ba-p/1257363)

Or 

Here : [Azure Active Directory (Azure AD) Graph to Microsoft Graph migration FAQ](https://docs.microsoft.com/en-us/graph/migrate-azure-ad-graph-faq)

Sorry if my bug request is not as expected, it's my first. 

### Steps to Reproduce

1. Follow guide at https://www.bookstackapp.com/docs/admin/third-party-auth/#azuread-microsoft
2. STEP 8  --> the ‘Azure Active Directory Graph’ is grayed out

### Expected Behaviour

Ability to enable the ‘Azure Active Directory Graph'

### Screenshots or Additional Context

![image](https://user-images.githubusercontent.com/92555922/139533778-2ec48133-5a8e-4a19-a47e-048d07e7c0e5.png)

### Exact BookStack Version

v21.10.1

### Hosting Environment

Docker version 20.10.9, build c2ea9bc

Bookstack Docker image : Linuxserver.io version:- v21.10.1-ls167 Build-date:- 2021-10-27T14:38:02+02:00

**Comment by ssddanbrown** [comment]
Thanks for raising @Risc-Skydoo.

---

### Notes

- Potentially related upstream changes to handle this: https://github.com/SocialiteProviders/Providers/commit/a92e81b64afb0d1593d72d6e881aa591c4537f6e
  - These changes were in a major release for the package: https://github.com/SocialiteProviders/Microsoft-Azure/releases

**Comment by ssddanbrown** [comment]
Applied 5c834f24a6e2b933236cbabc4484f00c686dc05c to update to Microsoft graph instead of AzureAD graph.

Tested using old AzureAD graph usage for backwards-compatibility, did not seem to break things. Could not test with conditional access though due to azure never enforcing it no matter what I attempted.

Will be part of next patch release. 

Need to update docs upon release to reflect new API configuration
